routes/web.php

```diff
@@ -358,9 +361,25 @@ function ($user, $password) {
 
 Route::post('/games/ptd/createGiveaway.php', [GiveawayController::class, 'create'])->middleware('auth');
 
-// Mystery Gift
-Route::get('/games/ptd/dailyCode.php', function () {return redirect('/p/mystery-gift.html');});
+// Daily Gift
 
+Route::get('/games/ptd/dailyGift.php', function () {
+    return view('dailyGift', ['save' => Auth::user()->selectedSave(), 'user' => Auth::user(), 'dateCheck' => Carbon::parse(Auth::user()->last_used_dg)->addDay()->isBefore(Carbon::now('UTC'))]);
```

Comment:
I assume the dateCheck doesn't work if the date is null and thus you have the null check (in other files), if that is the case, why not have the date check handle null with a check

Also I am unsure if this is the case but I think timestamp columns are autocast to Carbon so the parse might be redundant

Reply:
you mean check for the null in the function?

and the other reason for the parse was for the ->addDay->isBefore(...
so is it really redundant?

Reply:
Yes, you can check ahead of time in the route function

What I mean is, parsing shouldn't be necessary, timestamp columns in the db should be auto-converted to the carbon interface if I am not mistaken

Reply:
well when I try to do something like:
$dateCheck = Auth::user()->last_used_gc->addDay()->isBefore(Carbon::now('UTC'));
instead of 
$dateCheck = Carbon::parse(Auth::user()->last_used_gc)->addDay()->isBefore(Carbon::now('UTC'));
it gives me an error of 'Call to a member function addDay() on string' meaning that the function doesn't exist. meaning I have to user Carbon::parse(...

Reply:
Ah I see, this is the confusion https://laravel.com/docs/10.x/eloquent-mutators#date-casting if it is not created_at or updated_at, you need to manually tell it to cast "x" column to Carbon interface, so Carbon::parse can remain or you can add it as an auto-cast, up to you, idm